database.c

```diff
@@ -487,12 +515,21 @@ void save_to_DB(void)
 		return;
 	}
 
+	// Update individual queryIDs in the queries struct
+	sqlite3_int64 lastID = last_ID_in_DB();
+	for(i=0; i < total; i++)
+	{
+		// Subtract i from coutners-queries-1 as the database
+		// loop goes only until i < counters->queries
+		queries[(counters->queries-1)-i].db = lastID-i;
```

Comment:
This does not consider private queries,  queries too new (see lines 414-419), or save errors (if less than 3).

Maybe it would be better to get the last ID in the database before going into the main loop, and then setting the ID as each query is entered into the database?